MovementInputSystem.cs

```diff
@@ -0,0 +1,341 @@
+    {
+        if (other.CompareTag("Accelerator"))
+        {
+            StartCoroutine(ApplySpeedBoost());
+            // You may want to disable the accelerator object after it's been used.
+            other.gameObject.SetActive(false);
+        }
+    }
+
+    private IEnumerator ApplySpeedBoost()
+    {
+        if (!isSpeedBoosted)
+        {
+            // Apply speed boost
+            forwardSpeed *= speedBoostMultiplier;
+            backwardSpeed *= speedBoostMultiplier;
+            rotationSpeed *= speedBoostMultiplier;
+            turnSpeed *= speedBoostMultiplier;
+            rotateAlongZSpeed *= speedBoostMultiplier;
+
+            isSpeedBoosted = true;
+
+            // Wait for the duration of the speed boost
+            yield return new WaitForSeconds(speedBoostDuration);
+
+            // Remove speed boost
+            forwardSpeed /= speedBoostMultiplier;
+            backwardSpeed /= speedBoostMultiplier;
+            rotationSpeed /= speedBoostMultiplier;
+            turnSpeed /= speedBoostMultiplier;
+            rotateAlongZSpeed /= speedBoostMultiplier;
+
+            isSpeedBoosted = false;
+        }
+    }
+
+    private void Awake()
+    {
+        ship = GetComponent<Rigidbody>();
+
+        // Initialize fuel level
+        //currentFuelLevel = maxFuelCapacity;
+        fuelSystem = GetComponent<FuelSystem>(); // Get the FuelSystem component attached to the player ship
+
+
+        // Initialize movement actions
+        movementActions = new MovementActions();
+        movementActions.Ship.Forward.started += StartMovingForward;
+        movementActions.Ship.Forward.canceled += StopMovingForward;
+        movementActions.Ship.Back.started += StartMovingBackward;
+        movementActions.Ship.Back.canceled += StopMovingBackward;
+        movementActions.Ship.RotateLeft.started += StartRotatingLeft;
+        movementActions.Ship.RotateLeft.canceled += StopRotatingLeft;
+        movementActions.Ship.RotateRight.started += StartRotatingRight;
+        movementActions.Ship.RotateRight.canceled += StopRotatingRight;
+        movementActions.Ship.RotateUp.started += StartTurningUp;
+        movementActions.Ship.RotateUp.canceled += StopTurningUp;
+        movementActions.Ship.RotateDown.started += StartTurningDown;
+        movementActions.Ship.RotateDown.canceled += StopTurningDown;
+        movementActions.Ship.RotateAlongZLeft.started += StartRotatingAlongZLeft;
+        movementActions.Ship.RotateAlongZLeft.canceled += StopRotatingAlongZLeft;
+        movementActions.Ship.RotateAlongZRight.started += StartRotatingAlongZRight;
+        movementActions.Ship.RotateAlongZRight.canceled += StopRotatingAlongZRight;
+
+        // Start the fuel consumption coroutine
+        StartCoroutine(ConsumeFuel());
+    }
+
+    private void OnEnable()
+    {
+        movementActions.Enable();
+    }
+
+    private void OnDisable()
+    {
+        movementActions.Disable();
+    }
+
+    private IEnumerator ConsumeFuel()
```

Comment:
Should not be done within player movement, instead FuelSystem should listen to any movement from the MovementSystem and react accordingly. This here is high coupling, which is a bad practice